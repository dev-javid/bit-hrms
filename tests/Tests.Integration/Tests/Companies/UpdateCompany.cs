namespace Tests.Integration.Tests.Companies
{
    public class UpdateCompany(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/companies/1";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Update_Company()
        {
            await LoginAsSuperAdminAsync();
            await FeedDataAsync("Tests/Companies/UpdateCompany.sql");

            var command = new
            {
                Name = "New Company Name",
                AdministratorName = "New Administrator Name",
                Address = "Some Other Random Place on the Planet Earth",
                PhoneNumber = "1234567890",
            };

            HttpResponseMessage response;
            using (StringContent content = new StringContent(JsonSerializer.Serialize(command), Encoding.UTF8, JsoneMediaType))
            {
                response = await Client.PutAsync(Route, content);
            }

            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result);
        }
    }
}
