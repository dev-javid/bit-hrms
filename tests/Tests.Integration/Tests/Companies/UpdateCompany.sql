INSERT INTO public.companies (id, name, email, phone_number, financial_month, administrator_name, weekly_off_days, address, is_deleted, created_by, created_on)
VALUES 
(1,  'Test Company 1',  'test1@company.com', '1234567890', 1, 'Admin 1', ARRAY['Sunday'], 'Test Address 1', false, 1, now());
