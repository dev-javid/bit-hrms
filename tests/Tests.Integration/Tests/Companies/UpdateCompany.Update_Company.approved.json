{
    "weeklyOffDays": [
        "Sunday"
    ],
    "companyId": 999,
    "name": "New Company Name",
    "phoneNumber": "1234567890",
    "financialMonth": 3,
    "administratorName": "New Administrator Name",
    "email": "company1@example.com",
    "address": "Some Other Random Place on the Planet Earth"
}